### PR TITLE
PyPlot legends now show marker types.

### PR DESCRIPTION
**Pull request by JackDevine**

### CODE REVIEW

**Comment by JackDevine** [comment]
Fixes #911 

The new behaviour looks like this:
```julia
julia> p1 = plot(rand(10,4), markershape = :auto, linetype = :scatter)

julia> p2 = plot(rand(10,4), markershape = :auto)

julia> p3 = plot(rand(10,4))

julia> plot(p1, p2, p3)
```
new_style

**Comment by daschw** [comment]
Looks, great!
Is it also possible to add the markerstroke in `p2`?

**Comment by JackDevine** [comment]
```julia
julia> plot(rand(10, 4), markershape = :hexagon,
                  markersize = 20,
                  markeralpha = 0.6,
                  markercolor = :green,
                  markerstrokewidth = 3,
                  markerstrokealpha = 0.2,
                  markerstrokecolor = :black,
                  markerstrokestyle = :dot)
```
figure_1

So it doesn't really nail the subtelties of markerstrokes. Although, I am not sure if that is what you were asking.

**Comment by daschw** [comment]
Yes, that was what I was asking. In `p2 = plot(rand(10,4), markershape = :auto)` the markers in the plot have a border and the ones in the legend do not.

File: src/backends/pyplot.jl [hunk 1]
```diff
@@ -1183,6 +1183,12 @@ function py_add_legend(plt::Plot, sp::Subplot, ax)
                         color = py_color(_cycle(series[:fillcolor],1)),
                         linewidth = py_dpi_scale(plt, 4)
                     )
+                elseif series[:seriestype] == :path
+                    PyPlot.plt[:Line2D]((0,1),(0,0),
+                        color = py_color(_cycle(series[:fillcolor],1)),
+                        linewidth = py_dpi_scale(plt, 1),
+                        marker = py_marker(series[:markershape])
```
Comment (daschw) [inline src/backends/pyplot.jl]: maybe adding
```julia
edgecolor = py_markerstrokecolor(series)
facecolor = py_markercolor(series)
```
could fix the marker colors here. I'm not sure, though.

Comment (JackDevine) [reply src/backends/pyplot.jl]: Hmm, I tried these and I get the error:
```
Error showing value of type Plots.Plot{Plots.PyPlotBackend}:
ERROR: PyError (ccall(@pysym(:PyObject_Call), PyPtr, (PyPtr, PyPtr, PyPtr), o, arg, kw)) <type 'exceptions.AttributeError'>
AttributeError(u'Unknown property edgecolor',)
  File "/Users/jackdevine/.julia/v0.6/Conda/deps/usr/lib/python2.7/site-packages/matplotlib/lines.py", line 431, in __init__
    self.update(kwargs)
  File "/Users/jackdevine/.julia/v0.6/Conda/deps/usr/lib/python2.7/site-packages/matplotlib/artist.py", line 885, in update
    for k, v in props.items()]
  File "/Users/jackdevine/.julia/v0.6/Conda/deps/usr/lib/python2.7/site-packages/matplotlib/artist.py", line 878, in _update_property
    raise AttributeError('Unknown property %s' % k)
```
"Unknown property edgecolor"

I think that the problem is that I am using `PyPlot.plt[:Line2D]`, maybe I should try using something more specific.

Comment (daschw) [reply src/backends/pyplot.jl]: markeredgecolor and markerfacecolor could work with Line2D, cf. https://matplotlib.org/devdocs/api/_as_gen/matplotlib.lines.Line2D.html

Comment (JackDevine) [reply src/backends/pyplot.jl]: Thanks! That works :)

Comment (daschw) [reply src/backends/pyplot.jl]: Great!

**Comment by mkborregaard** [comment]
Travis error is unrelated

**Comment by mkborregaard** [comment]
Thanks!

**Pull request merged**